js/ui/drop_down_button.js

```diff
@@ -490,19 +487,19 @@ const DropDownButton = Widget.inherit({
 
         this._buttonGroup = this._createComponent($buttonGroup, ButtonGroup, this._buttonGroupOptions());
 
-        const buttonElements = this._buttonGroup.$element().find('.dx-button').toArray();
+        this._$buttonElements = this._buttonGroup.$element().find('.dx-button');
 
-        this._ariaExpandedElements = [ ...buttonElements, this.$element() ];
-
-        this._buttonGroup.registerKeyHandler('downArrow', this._upDownKeyHandler.bind(this));
-        this._buttonGroup.registerKeyHandler('tab', this._tabHandler.bind(this));
-        this._buttonGroup.registerKeyHandler('upArrow', this._upDownKeyHandler.bind(this));
-        this._buttonGroup.registerKeyHandler('escape', this._escHandler.bind(this));
+        this._buttonGroup._buttonsCollection.registerKeyHandler('downArrow', this._upDownKeyHandler.bind(this));
```

Comment:
`_buttonsCollection` is a private api of the buttonGroup -> it's not a good idea to use it outside of the buttonGroup.
Cannot we don't do anything with `esc` and `tab` and just forcibly focus $toggleButton in `_upDownKeyHandler`?

Reply:
We can don't do anything with esc and tab key handlers.

But forcibly focus $toggleButton in _upDownKeyHandler is so strange solution for me because focus will be changed twice: set to the action button in buttonsCollection handler, then set back to the toggle button in buttonGroup handler.

Moreover, I think it's right to override the parent behavior in the current component, because this avoids side effects such as setting focus to the first element.

Also I think you are right because using a private api is discouraged. But there is no public api in the current component. And the private api is used by us everywhere in our library.

if you are still against using _buttonsCollection so let's have a voice discussion.

Reply:
We discussed it in voice. Decided not to revert this changed and leave the previous behavior.

Reply:
Reverted here: [08b3ae3acf750d1245b3bd08f0c6e468113ec5c1].